include/mls/credential.h

```diff
@@ -38,6 +38,29 @@ struct BasicCredential
   static const CredentialType type;
 };
 
+struct X509Credential
+{
+  struct CertData
+  {
+    bytes data;
+
+    TLS_SERIALIZABLE(data);
+    TLS_TRAITS(tls::vector<2>)
+  };
+
+  X509Credential() {}
+
+  explicit X509Credential(std::vector<CertData> der_chain_in);
+
+  std::vector<CertData> der_chain;
+  SignaturePublicKey public_key;
```

Comment:
1. This should either be const or private and exposed via a method.
2. We should expose the signature algorithm here and verify it in `KeyPackage::verify()`.  Please file an issue to do this in a follow-on.